docs/reporting/reqnroll-formatters.md

```diff
@@ -0,0 +1,121 @@
+
+```{note}
+For more information about Cucumber Messages and the tooling that consumes them, please see the [Cucumber Messages](https://github.com/cucumber/messages) page on Github.
+```
+
+The Cucumber Messages formatter can be enabled with a `message` section in the configuration file within `formatters`. By default, it generates a `reqnroll_report.ndjson` file to the project output folder (e.g., `bin/Debug/net8.0/`), but this can be configured using the `outputFilePath` setting.
+
+The following configuration file enables the Cucumber Messages formatter and generates the report to the `report/cucumber_messages.ndjson` file within the project output folder.
+
+```{code-block} json
+:caption: reqnroll.json
+{
+  "$schema": "https://schemas.reqnroll.net/reqnroll-config-latest.json",
+
+    "formatters": {
+        "message" : { "outputFilePath" : "report/cucumber_messages.ndjson" }
+    }
+}
+```
+
+## Creating a custom formatter
+
+```{warning}
+The formatter infrastructure is new, therefore the interface details and the implementation rules might change significantly even in minor version changes. 
+```
+
+To create a custom formatter, you need to create a Reqnroll [Runtime plugin](../extend/plugins.md#runtime-plugins), create a class that implements the `ICucumberMessageFormatter` interface and register it to the test run (global) DI container with a new name. This name must be the same as returned by the `ICucumberMessageFormatter.Name` property of the formatter implementation.
+
+The following example registers the `CustomFormatter` with the name `custom`.
+
+```{code-block} c#
+using Reqnroll.Formatters;
+using Reqnroll.Plugins;
+using Reqnroll.UnitTestProvider;
+using ReqnrollFormatters.Custom;
+
+[assembly: RuntimePlugin(typeof(CustomFormatterPlugin))]
+
+namespace ReqnrollFormatters.Custom;
+
+public class CustomFormatterPlugin : IRuntimePlugin
+{
+    public void Initialize(RuntimePluginEvents runtimePluginEvents, RuntimePluginParameters runtimePluginParameters, UnitTestProviderConfiguration unitTestProviderConfiguration)
+    {
+        runtimePluginEvents.RegisterGlobalDependencies += (_, args) =>
+        {
+            args.ObjectContainer.RegisterTypeAs<CustomFormatter, ICucumberMessageFormatter>("custom");
+        };
+    }
+}
+```
+
+The created formatter can be enabled with a `custom` section in the configuration file within `formatters`.
+
+
+```{code-block} json
+:caption: reqnroll.json
+{
+  "$schema": "https://schemas.reqnroll.net/reqnroll-config-latest.json",
+
+    "formatters": {
+        "custom" : { "outputFilePath" : "custom_report.txt" }
+    }
+}
+```
+
+For a complete example that contains a custom formatter, please check our [Custom Formatter Test Project](https://github.com/reqnroll/Reqnroll.ExploratoryTestProjects/tree/main/ReqnrollFormatters/ReqnrollFormatters.Custom).
```

Comment:
Nice work on the sample project!  
The sample project's example of using the default formatters uses the entry "messages"; needs to be changed to 'message'.